MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/HistoryController.java

```diff
@@ -67,41 +76,60 @@ public void customizeBinding(WebDataBinder binder) {
-            model.addAttribute("historiesSession", pdfCreatorService.getHistoriesByUserId());
-            return "history";
+    public String getHistories(Model model, HttpSession session,
+                               @RequestParam(value = "pageNumber", required = false, defaultValue = "1") int pageNumber,
+                               @RequestParam(value = "pageSize", required = false, defaultValue = "25") int pageSize) {
+
+        HistorySearchDTO searchParams = session.getAttribute("historySearchDTO") == null ? new HistorySearchDTO()
+                : (HistorySearchDTO) session.getAttribute("historySearchDTO");
+        model.addAttribute("historySearchDTO", searchParams);
+        model.addAttribute("categories", productPageService.getAllCategories(userService.getCurrentUser().getId()));
+
+        Page<History> histories = historyService.getHistorySearchPage(searchParams, pageNumber, pageSize);
+        if (histories.getTotalElements() > 0) {
+            model.addAttribute("historiesSession", histories.getContent());
+            model.addAttribute("historiesPage", histories);
+            model.addAttribute("pageSize", pageSize);
+            model.addAttribute("beginIndex", 1);
+            model.addAttribute("endIndex", histories.getTotalPages());
+            model.addAttribute("currentIndex", histories.getNumber() + 1);
         }
-        return "emptyHistory";
+
+        return "history";
     }
 
     /**
-     * Handles requests for getting all history records for current user by given search criterias
+     * Handles requests for search history records by given search criterias
      *
-     * @param model - a map that will be handed off to the view for rendering the data to the client
-     * @return string with appropriate view name
+     * @param searchParams - DTO with search form values
+     * @return redirect to regular history page
      */
-    @PostMapping
-    public String searchHistories(Model model, @ModelAttribute("historySearchDTO") HistorySearchDTO searchParams, BindingResult result) {
-        model.addAttribute("historySearchDTO", result.hasErrors() ? new HistorySearchDTO() : searchParams);
-        model.addAttribute("categories", productPageService.getAllCategories(userService.getCurrentUser().getId()));
 
-        List<History> histories = historyService.getByUserIdAndSearchParams(searchParams);
-        if (CollectionUtils.isNotEmpty(histories)) {
-            model.addAttribute("histories", histories);
-            model.addAttribute("historiesSession", pdfCreatorService.getByUserIdAndSearchParams(searchParams));
+    @PostMapping
+    public String searchHistories(HttpSession session, @ModelAttribute("historySearchDTO") HistorySearchDTO searchParams,
+                                  BindingResult result) {
+        if (result.hasErrors()) {
+            session.setAttribute("historySearchDTO", new HistorySearchDTO());
         }
+        return "redirect:/history";
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "redirect:/history" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)